lib/binary-breeder.js

```diff
@@ -1,76 +1,85 @@
 
-    return killRandomChildren(newChildren, numOffspring);
+        return breeder._killRandomChildren(newChildren, optionsToUse.numOffspring);
+    },
 
-    function validateParentChromosomes(parentChromosomes) {
-        if(!Array.isArray(parentChromosomes)) {
+    _getDefaultOptions(options) {
+        return {
+            mutationChance: (options || {}).mutationChance || 0.01,
+            numOffspring: (options || {}).numOffspring || 10
+        };
+    },
+
+    _validateParentChromosomes(parentChromosomes) {
+        if (!Array.isArray(parentChromosomes)) {
             throw new Error('Parent chromosomes must be provided in an array.');
         }
 
-        if(parentChromosomes.length !== 2) {
+        if (parentChromosomes.length !== 2) {
             throw new Error('Exactly two parent chromosomes should be specified.');
         }
 
-        if(typeof parentChromosomes[0] !== 'string' || typeof parentChromosomes[1] !== 'string') {
+        if (typeof parentChromosomes[0] !== 'string' || typeof parentChromosomes[1] !== 'string') {
             throw new Error('Both parent chromosomes must be strings.');
         }
 
-        if(parentChromosomes[0].length !== parentChromosomes[1].length) {
-            throw new Error('Both chromosomes must be the same length');
+        if (parentChromosomes[0].length !== parentChromosomes[1].length) {
+            throw new Error('Both chromosomes must be the same length.');
         }
 
-        var only0and1Regex = /^[01]+$/g;
-        if(parentChromosomes[0].match(only0and1Regex) === null || parentChromosomes[1].match(only0and1Regex) === null) {
-            throw new Error('Both chromosomes may only contain only 0s and 1s.')
+        if (parentChromosomes[0].match(only0and1Regex) === null || parentChromosomes[1].match(only0and1Regex) === null) {
+            throw new Error('Both chromosomes may only contain only 0s and 1s.');
         }
-    }
+    },
 
-    function killRandomChildren(newChildren, numRequestedChildren) {
-        var numGenerated = newChildren.length;
-        for(var i = 0; i < numGenerated - numRequestedChildren; ++i) {
-            var randomIndexToKill = Random.integer(0, newChildren.length - 1);
+    _killRandomChildren(newChildren, numRequestedChildren) {
+        const numGenerated = newChildren.length;
+        for (let i = 0; i < numGenerated - numRequestedChildren; ++i) {
+            const randomIndexToKill = Random.integer(0, newChildren.length - 1);
             newChildren.splice(randomIndexToKill, 1);
         }
         return newChildren;
-    }
+    },
 
-    function mutate(chromosome, mutationChance) {
-        var mutatedChromosome = '';
-        for(var i = 0; i < chromosome.length; ++i) {
-            if(Random.real(0, 0.999999) < mutationChance) {
+    _mutate(chromosome, mutationChance) {
+        let mutatedChromosome = '';
+        for (let i = 0; i < chromosome.length; ++i) {
+            if (Random.real(0, 0.999999) < mutationChance) {
```

Comment:
I can see how that'd be interesting, a weighted mutation chance depending on the gene in question.  In a future revision perhaps.